x/ibc/04-channel/keeper/packet.go

```diff
@@ -193,83 +187,80 @@ func (k Keeper) SendPacket(
-	acknowledgement []byte,
-	portCapability sdk.CapabilityKey,
-) (exported.PacketI, error) {
+) (types.Packet, error) {
 
 	channel, found := k.GetChannel(ctx, packet.GetDestPort(), packet.GetDestChannel())
 	if !found {
-		return nil, types.ErrChannelNotFound(k.codespace, packet.GetDestPort(), packet.GetDestChannel())
+		return types.Packet{}, types.ErrChannelNotFound(k.codespace, packet.GetDestPort(), packet.GetDestChannel())
 	}
 
 	if channel.State != types.OPEN {
-		return nil, types.ErrInvalidChannelState(
+		return types.Packet{}, types.ErrInvalidChannelState(
 			k.codespace,
 			fmt.Sprintf("channel state is not OPEN (got %s)", channel.State.String()),
 		)
 	}
 
-	if !k.portKeeper.Authenticate(portCapability, packet.GetDestPort()) {
-		return nil, errors.New("port is not valid")
-	}
-
 	// packet must come from the channel's counterparty
 	if packet.GetSourcePort() != channel.Counterparty.PortID {
-		return nil, types.ErrInvalidPacket(
+		return types.Packet{}, types.ErrInvalidPacket(
 			k.codespace,
 			fmt.Sprintf("packet source port doesn't match the counterparty's port (%s ≠ %s)", packet.GetSourcePort(), channel.Counterparty.PortID),
 		)
 	}
 
 	if packet.GetSourceChannel() != channel.Counterparty.ChannelID {
-		return nil, types.ErrInvalidPacket(
+		return types.Packet{}, types.ErrInvalidPacket(
 			k.codespace,
 			fmt.Sprintf("packet source channel doesn't match the counterparty's channel (%s ≠ %s)", packet.GetSourceChannel(), channel.Counterparty.ChannelID),
 		)
 	}
 
 	connectionEnd, found := k.connectionKeeper.GetConnection(ctx, channel.ConnectionHops[0])
 	if !found {
-		return nil, connection.ErrConnectionNotFound(k.codespace, channel.ConnectionHops[0])
+		return types.Packet{}, connection.ErrConnectionNotFound(k.codespace, channel.ConnectionHops[0])
 	}
 
 	if connectionEnd.State != connection.OPEN {
-		return nil, connection.ErrInvalidConnectionState(
+		return types.Packet{}, connection.ErrInvalidConnectionState(
 			k.codespace,
 			fmt.Sprintf("connection state is not OPEN (got %s)", connectionEnd.State.String()),
 		)
 	}
 
 	if uint64(ctx.BlockHeight()) >= packet.GetTimeoutHeight() {
-		return nil, types.ErrPacketTimeout(k.codespace)
+		return types.Packet{}, types.ErrPacketTimeout(k.codespace)
 	}
 
 	if !k.connectionKeeper.VerifyMembership(
 		ctx, connectionEnd, proofHeight, proof,
 		types.PacketCommitmentPath(packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence()),
-		packet.GetData(), // TODO: hash data
+		packet.GetCommitment(), // TODO: hash data
 	) {
-		return nil, errors.New("couldn't verify counterparty packet commitment")
+		return types.Packet{}, errors.New("couldn't verify counterparty packet commitment")
 	}
 
-	if len(acknowledgement) > 0 || channel.Ordering == types.UNORDERED {
-		k.SetPacketAcknowledgement(
-			ctx, packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence(),
-			acknowledgement, // TODO: hash ACK
-		)
-	}
+	// XXX: move to WriteAcknowledgement
```

Comment:
Has this been done? This is part of ADR 15 I believe.

Reply:
This work has been completed in a separate PR? Can this comment be resolved?